File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_co2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your CO2 Bricklet

ipcon = IPConnection.new # Create IP connection
co2 = BrickletCO2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register CO2 concentration callback
co2.register_callback(BrickletCO2::CALLBACK_CO2_CONCENTRATION) do |co2_concentration|
  puts "CO2 Concentration: #{co2_concentration} ppm"
end

# Set period for CO2 concentration callback to 1s (1000ms)
# Note: The CO2 concentration callback is only called every second
#       if the CO2 concentration has changed since the last call!
co2.set_co2_concentration_callback_period 1000

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletCO2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletCO2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your CO2 Bricklet

// Callback function for CO2 concentration callback
function cb_co2Concentration($co2_concentration)
{
    echo "CO2 Concentration: $co2_concentration ppm\n";
}

$ipcon = new IPConnection(); // Create IP connection
$co2 = new BrickletCO2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register CO2 concentration callback to function cb_co2Concentration
$co2->registerCallback(BrickletCO2::CALLBACK_CO2_CONCENTRATION, 'cb_co2Concentration');

// Set period for CO2 concentration callback to 1s (1000ms)
// Note: The CO2 concentration callback is only called every second
//       if the CO2 concentration has changed since the last call!
$co2->setCO2ConcentrationCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
